app/javascript/packs/init_create_category.js

```diff
@@ -23,114 +25,256 @@ const inItCreateCategory = () => {
   //     console.log(e.srcElement)
   //   })
   // })
+  
 
   // conditional to stop error when not on list#show page
   if (categoryCards.length > 1) {
     const id = getListId.dataset.id;
     // find the last category id
     const lastId = getListId.dataset.lastid;
-    console.log(lastId);
+    let lastIdInt = parseInt(lastId);
 
     categoryCards[0].addEventListener('click', (e) => {
       apiCall('Meat & Fish', '#d8315b', '#ffecf0', id)
       categoryModal.style.display = 'none';
       
       // call the newCategoryDiv funcation and save it to div.
-      const div = newCategoryDiv('Meat & Fish', '#d8315b', lastId);
-      console.log(div);
-      console.log(id);
+      const div = newCategoryDiv('Meat & Fish', '#d8315b', lastIdInt);
       categoryContainer.insertAdjacentHTML("beforeend", div);
+
+      // reload this page
+      setTimeout(function (){
+
+        location.reload();
+      
+      }, 10);
     })
 
     categoryCards[1].addEventListener('click', (e) => {
       apiCall('Fruit & Veg', '#50723c', '#e1ffd0', id)
       console.log(id)
       categoryModal.style.display = 'none';
+
+      const div = newCategoryDiv('Fruit & Veg', '#50723c', lastIdInt);
+      categoryContainer.insertAdjacentHTML("beforeend", div);
+
+      setTimeout(function (){
+
+        location.reload();
+      
+      }, 10);
     })
 
     categoryCards[2].addEventListener('click', (e) => {
       apiCall('Snacks', '#f2a540', '#fff0dc', id)
       console.log(id)
       categoryModal.style.display = 'none';
+
+      const div = newCategoryDiv('Snacks', '#f2a540', lastIdInt);
+      categoryContainer.insertAdjacentHTML("beforeend", div);
+
+      setTimeout(function (){
+
+        location.reload();
+      
+      }, 10);
```

Comment:
could we consider making the reload time shorter?

Reply:
perhaps. It's not doing what I had wanted it to do. I thought perhaps the page was reloading too quickly... OK. now I'm typing this, I realise it sounds kinda stupid.